objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestUploadAndDeleteQuery.java

```diff
@@ -0,0 +1,612 @@
+        core.start(10);
+        
+        vantiq = new io.vantiq.client.Vantiq(testVantiqServer);
+        vantiq.setAccessToken(testAuthToken);
+        
+        // Add files to be deleted from VANTIQ later
+        vantiqUploadFiles.add(IMAGE_NAME_1);
+        vantiqUploadFiles.add(IMAGE_NAME_2);
+        vantiqUploadFiles.add(IMAGE_NAME_3);
+        vantiqUploadFiles.add(IMAGE_NAME_4);
+        vantiqUploadFiles.add(IMAGE_NAME_5);
+    }
+    
+    @Before
+    public void addLocalTestImages() throws IOException {
+        new File(OUTPUT_DIR).mkdirs();
+        
+        byte[] testImageBytes = getTestImage();
+        InputStream in = new ByteArrayInputStream(testImageBytes);
+        BufferedImage testImageBuffer = ImageIO.read(in);
+        
+        File testFile = new File(OUTPUT_DIR + File.separator + "2019-02-05--02-35-10.jpg");
+        ImageIO.write(testImageBuffer, "jpg", testFile);
+        
+        testFile = new File(OUTPUT_DIR + File.separator + "2019-02-05--02-35-13.jpg");
+        ImageIO.write(testImageBuffer, "jpg", testFile);
+        
+        testFile = new File(OUTPUT_DIR + File.separator + "2019-02-05--02-35-16.jpg");
+        ImageIO.write(testImageBuffer, "jpg", testFile);
+        
+        testFile = new File(OUTPUT_DIR + File.separator + "2019-02-05--02-35-19.jpg");
+        ImageIO.write(testImageBuffer, "jpg", testFile);
+        
+        testFile = new File(OUTPUT_DIR + File.separator + "2019-02-05--02-35-22.jpg");
+        ImageIO.write(testImageBuffer, "jpg", testFile);
+    }
+    
+    @AfterClass
+    public static void tearDown() {
+        core.stop();
+    }
+    
+    @After
+    public void deleteFromVantiq() throws InterruptedException {
+        for (int i = 0; i < vantiqUploadFiles.size(); i++) {
+            Thread.sleep(1000);
+            vantiq.deleteOne("system.documents", vantiqUploadFiles.get(i), new BaseResponseHandler() {
+
+                @Override
+                public void onSuccess(Object body, Response response) {
+                    super.onSuccess(body, response);
+                }
+
+                @Override
+                public void onError(List<VantiqError> errors, Response response) {
+                    super.onError(errors, response);
+                }
+
+            });
+        }
+        
+        File d = new File(OUTPUT_DIR);
+        if (d.exists()) {
+            deleteDirectory(OUTPUT_DIR);
+        }
+    }
+    
+    @Test
+    public void testInvalidImageUploadParameters() throws InterruptedException {
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        
+        // Forgetting to include imageDir
+        Map<String, Object> request = new LinkedHashMap<>();
+        request.put("imageName", "all");
+        core.uploadLocalImages(request, null);
+                
+        // Checking that all images were uploaded to VANTIQ
```

Comment:
"all the images are still there?" or aren't there?  Didn't these get inserted in setup()